connect/runtime/src/main/java/org/apache/kafka/connect/runtime/health/ConnectClusterStateImpl.java

```diff
@@ -42,13 +45,25 @@ public ConnectClusterStateImpl(HerderProvider herderProvider) {
 
     @Override
     public Collection<String> connectors() {
+        final CountDownLatch connectorsLatch = new CountDownLatch(1);
         final Collection<String> connectors = new ArrayList<>();
         herderProvider.get().connectors(new Callback<java.util.Collection<String>>() {
             @Override
             public void onCompletion(Throwable error, Collection<String> result) {
                 connectors.addAll(result);
+                connectorsLatch.countDown();
             }
         });
+        try {
+            // Should take no more than 10 seconds to get a list of connectors. Any longer than this
+            // and it's very likely something is wrong.
+            boolean success = connectorsLatch.await(10, TimeUnit.SECONDS);
```

Comment:
Have you compared this value with `ConnectorsResource.REQUEST_TIMEOUT_MS`? I know the other one is taking into account rebalances. But how much do we care here? The diff between 90sec and 10sec seems significant.

Reply:
It looks like the `DistributedHerderRequest` that gets created to fulfill a call to `DistributedHerder.connectors(...)` should complete quickly and won't be blocked even during rebalance. I'm unclear if other requests that are queued up before it could cause a call to `DistributedHerder.connectors(...)` to still take longer than ten seconds but I'm leaning toward no...

Reply:
Timeout to read the log and get a snapshot is 3sec. I agree. I think we are good.